tinyagentos/code_analyzer.py

```
@@ -0,0 +1,477 @@
+    exfiltration pattern when a network call (fetch/XHR/WebSocket) shows up
+    nearby in the same file, implying the read value is being shipped out.
+    This is a same-file line-proximity heuristic
+    (`_STORAGE_EXFIL_WINDOW` lines in either direction), not
+    dataflow analysis -- it does not verify the read value is actually
+    what's sent over the network, so an unrelated storage read and network
+    call that both happen to live in the same small file can false-positive,
+    while a read/send split across two files is missed entirely.
+    """
+    findings: list[Finding] = []
+    lines = _iter_lines(content)
+    storage_lines = [i for i, l in enumerate(lines) if _STORAGE_READ_RE.search(l)]
```

Ruff (0.15.20)

[error] Ambiguous variable name: `l`

(E741)